cobigen/cobigen-core-parent/cobigen-core-systemtest/src/test/java/com/devonfw/cobigen/systemtest/GenerationTest.java

```diff
@@ -90,21 +90,25 @@ public void testGenerationWithExternalIncrements() throws Exception {
         List<IncrementTo> increments = cobigen.getMatchingIncrements(input);
         List<String> triggersIds = cobigen.getMatchingTriggerIds(input);
 
-        // assert
-        IncrementTo externalIncrement = null;
         // we try to get an increment containing external increments
         for (IncrementTo inc : increments) {
             if (inc.getId().equals("3")) {
-                externalIncrement = inc;
+                // We expect increment 3 to have an external increment 0 containing one template
+                assertThat(inc).isNotNull();
+                assertThat(inc.getDependentIncrements().get(0).getTemplates().size()).isEqualTo(1);
```

Comment:
Please use assertJ assertions to provide better error messages on failing tests.
Here you could use assertThat(Inc.getDependentIncrements().get(0).getTemplates()).hasSize(1)